packages/block-library/src/post-template/index.php

```diff
@@ -109,41 +109,3 @@ function render_legacy_query_loop_block( $attributes, $content, $block ) {
 	);
 	return render_block_core_post_template( $attributes, $content, $block );
 }
-
-/**
- * Complements the renaming of `Query Loop` to `Post Template`.
- * This ensures backwards compatibility for any users running the Gutenberg
- * plugin who have used Query Loop prior to its renaming.
- *
- * This can be removed when WordPress 5.9 is released.
- *
- * @see https://github.com/WordPress/gutenberg/pull/32514
- */
-function gutenberg_register_legacy_query_loop_block() {
```

Comment:
Like all blocks, these block-library files (and the legacy widget one) are synchronized automatically with core (copied as is). I feel we shouldn't create exceptions here.

Would you mind sharing the reasons you moved these files.

---
Aside: I also know that these files have a specific behavior in the plugin's build pipiline:

If a function doesn't have a `gutenberg_` prefix, it gets added automatically when building the Gutenberg plugin to avoid conflicts with core and also to allow these files to be copied into Core without the prefix. Meaning for this particular function, it seems the simple fix is just to remove the prefix here.

Reply:
I think that I may have misunderstood the need for the function. I interpreted `This ensures backwards compatibility for any users running the Gutenberg plugin who have used Query Loop prior to its renaming` as this is only needed in the plugin. My thinking was that the old style block never existed in Core, so this conversion would not be needed.

Thinking it through now, it's possible that a user had Gutenberg active before the blocks were renamed, and deactivated before the conversion could occur.

@youknowriad can you confirm that the0 notice in the plugin is currently triggered when the old format block renders?  If this is the case, I think we need to update this to something more permanent using a `__doing_it_wrong()` notice in Core. We can't know for sure that every post that contained the old style blocks get updated before 5.9 is released. That could be a different ticket though.

Reply:
I've reverted the change moving this function, updated the two related functions to include a `wp_` prefix, removed the note about removing them for now, and added `@since` and `@private` tags that are required for Core inline documentation standards.

Reply:
>  I interpreted This ensures backwards compatibility for any users running the Gutenberg plugin who have used Query Loop prior to its renaming as this is only needed in the plugin

Actually, you might be right here 😬 for some reason I thought this was the function used to register the block itself and not the compatibility code. @ntsekouras do you confirm that this code is not needed in Core?

(I'm sorry for the back and forth)

Reply:
>Thinking it through now, it's possible that a user had Gutenberg active before the blocks were renamed, and deactivated before the conversion could occur.

I guess this could be an edge case scenario. I tested with a jurassic site and the warning is shown. Should we handle this though, since in this scenario GB plugin would be deactivated. Should a user expect from a plugin to be working if they have disabled it? 

I guess it would be okay not be included in core, no?

Reply:
yes, it seems ok to move this code out of Core, I read it wrong in the first place.

Reply:
Personally, I think it's reasonable to say that the final state of a feature when it lands in WordPress itself may differ from the way it was previously in the plugin. But I also think it's fair to say that the plugin is responsible for handling those edge cases because the user had the plugin active to cause them in the first place.